app/public.js

```diff
@@ -70,13 +72,50 @@
         }
       }
     });
+    setRefreshImagesWatcher();
+  }
+
+  /**
+   * Refresh overlay
+   */
+  function refreshImagesInfo() {
+    hideImagesInfo();
+    showImagesInfo();
+  }
+
+  /**
+   * Set event listeners
+   */
+  function setRefreshImagesWatcher() {
```

Comment:
@tabanliviu MAINTAINABILITY - Please clear the timeout in case another event implements the same method.